### ISSUE TITLE
New errors with Selenium

### ISSUE THREAD

**Issue by kevinquillen** [geerlingguy/drupal-vm#418]
Fresh VM install, latest release of drupalvm. I have destroyed and rebuilt this VM twice using the same config from another project for Behat, I am getting this error when I run tests:

```
Could not open connection: Cannot find firefox binary in PATH. Make sure firefox is installed. OS appears to be: LINUX
Build info: version: '2.48.2', revision: '41bccdd', time: '2015-10-09 19:59:12'
System info: host: 'local', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.19.0-25-generic', java.version: '1.7.0_95'
Driver info: driver.version: FirefoxDriver
```

This is from the VM side. Selenium is running, everything seems right... is this a new bug or did I miss something?

**Comment by kevinquillen** [comment]
Changed behat.yml to leverage goutte to doublecheck... that works. 

What would have changed to cause this issue?

**Comment by geerlingguy** [comment]
This seems like it could be an upstream issue in `arknoll.selenium`: https://github.com/arknoll/ansible-role-selenium

**Comment by kevinquillen** [comment]
This sounds like a similar problem:

https://github.com/SeleniumHQ/selenium/issues/431

**Comment by kevinquillen** [comment]
Also:

https://github.com/arknoll/ansible-role-selenium/commit/7fa67818fc47146dc44589ad910b0c1e82f5459b

https://github.com/arknoll/ansible-role-selenium/pull/4

The versions in which I had Behat working would have been before this date
of change.

**Comment by kevinquillen** [comment]
This is the problem: 

https://github.com/arknoll/ansible-role-selenium/pull/4

I made that change to my local role, reloaded the VM with provision flag, and the 'install browser' task is now running. Firefox installed, and Selenium returns no errors.

**Comment by justinlevi** [comment]
@kevinquillen Can you elaborate on how you changed your local role? I'm also getting the same firefox binary error when trying to run a custom step. I'm really new to Ansible so probably missing something obvious.

**Comment by kevinquillen** [comment]
I actually edited the role directly to test and get around it.

**Comment by oxyc** [comment]
@justinlevi looking at the upstream patch a workaround would be to set the `selenium_install_firefox` variable to a string instead of a bool.

```yaml
selenium_install_firefox: "true"
```

Other option would be to "hack" Drupal VMs `requirements.yml` and `playbook.yml` by changing the `arknoll.selenium` dependency to `beetboxvm.selenium` instead.

- https://github.com/geerlingguy/drupal-vm/blob/master/provisioning/requirements.yml#L2
- https://github.com/geerlingguy/drupal-vm/blob/master/provisioning/playbook.yml#L51

**Comment by geerlingguy** [comment]
The upstream bug has been fixed and a new tag (1.2.0) is out, so this issue can be closed. However, there is still an open issue (which I have only a small bit of skin in the game for) concerning whether we want to switch to another selenium/browser driver setup, or do something else entirely... but there's an open ticket dealing with parts of that already, so closed!
